src/main/java/dobby/TaskList.java

```diff
@@ -0,0 +1,171 @@
+package dobby;
+
+import java.text.ParseException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+
+import dobby.task.Deadline;
+import dobby.task.Event;
+import dobby.task.Task;
+import dobby.task.TimedTask;
+import dobby.task.Todo;
+
+/**
+ * Stores the tasks and performs operations or returns details depending on function called
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
+
+    public TaskList() {
+        tasks = new ArrayList<Task>();
+    }
+
+    /**
+     * Create task with details from the input string and add to the tasks list
+     */
+    public void createFromStorage(String str) throws ParseException {
+        boolean isDone = str.charAt(4) == '\u2713';
+        Task task;
+        if (str.charAt(1) == 'T') { // TODO
+            String decription = str.substring(str.indexOf(' ') + 1);
+
+            task = new Todo(decription);
+            tasks.add(task);
+        } else if (str.charAt(1) == 'D') { // DEADLINE
+            String description = str.substring(str.indexOf(' ') + 1, str.indexOf("(by: ") - 1);
+            String by = str.substring(str.indexOf("(by: ") + 5, str.length() - 1); //Aug 28 2020 4:00 pm
+
+            int thirdIndex = by.indexOf(' ', 10);
+            String dateText = by.substring(0, thirdIndex);
+            String timeText = by.substring(thirdIndex + 1);
+            LocalDate date = LocalDate.parse(dateText, DateTimeFormatter.ofPattern("MMM d yyyy"));
+
+            task = new Deadline(description, timeText, date);
+            tasks.add(task);
+        } else { // EVENT
+            String description = str.substring(str.indexOf(' ') + 1, str.indexOf("(at: ") - 1);
+            String at = str.substring(str.indexOf("(at: ") + 5, str.length() - 1);
+            int thirdIndex = at.indexOf(' ', 10);
+
+            String dateText = at.substring(0, thirdIndex);
+            LocalDate date = LocalDate.parse(dateText, DateTimeFormatter.ofPattern("MMM d yyyy"));
+
+            String timeText = at.substring(thirdIndex + 1);
+            task = new Event(description, timeText, date);
+
+            tasks.add(task);
+        }
+
+        if (isDone) {
+            task.setDone();
+        }
+    }
+
+    public int getSize() {
+        return this.tasks.size();
+    }
+
+    public void addToList(Task task) {
+        this.tasks.add(task);
+    }
+
+    /**
+     * Returns a string format of all the tasks in the list
+     * @return String all tasks in the list in a particular format
+     */
+    public String getListedTasks() {
+        int i = 0;
```

Comment:
Your code looks good. I think the naming might could be made more readible here, so instead of using i, maybe it could be "index" instead, since I saw that you are using it to show the index of the list.

Reply:
Noted on all your comments. 
Thank you for taking the time to review my code and give your valuable insights!